src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let initialState = { ...state };
+
+  actions.forEach((action) => {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      initialState = { ...initialState, ...extraData };
+    } else if (type === 'removeProperties') {
+      initialState = { ...initialState };
```

Comment:
Creating a copy of `initialState` is unnecessary in the 'removeProperties' case because you are already making a copy when you delete the keys. You can remove this line.